dhis-2/dhis-api/src/main/java/org/hisp/dhis/expression/ExpressionParams.java

```diff
@@ -66,6 +66,8 @@
 @Builder( toBuilder = true )
 public class ExpressionParams
 {
+    public static final ExpressionParams DEFAULT_EXPRESSION_PARAMS;
```

Comment:
Are `ExpressionParams` truly immutable? I got the impression they have mutable state that changes during traversal. In that case a constant makes little sense and can cause different runs to affect each other.

Reply:
Yes, `ExpressionParams` are truly immutable. Every field except one is final and there is no good reason for that -- I will fix that in another PR. Once the instance is created all the collections are never modified, only read. (It would be nice if one could declare a collection in an object like this as unmodifiable -- do you know any reasonable way to do this?)

`ExpressionState` on the other hand is mutable and holds the current parsing state, but `ExpressionParams` describes the parameters used to invoke expression processing and is immutable.